ui-tests/cypress/integration/content/templates.spec.js

```diff
@@ -0,0 +1,91 @@
+import HomePage from '../../support/pageObjects/HomePage';
+import { htmlElements } from '../../support/pageObjects/WebElement';
+import { generateRandomId, generateRandomNumericId } from '../../support/utils';
+
+const openContentTemplatesPage = () => {
+  cy.visit('/');
+  let currentPage = new HomePage();
+  currentPage     = currentPage.getMenu().getContent().open();
+  return currentPage.openTemplates();
+};
+
+const addContentTemplate = template => cy.contentTemplatesController().then(controller => controller.addContentTemplate(template));
+const deleteContentTemplate = id => cy.contentTemplatesController().then(controller => controller.deleteContentTemplate(id));
+
+describe('Content Templates', () => {
+
+  let currentPage;
+  let template = {
+    contentType: 'BNR',
+    contentTypeText: 'Banner',
+    contentShape: '<div>test</div>'
+  };
+
+  beforeEach(() => {
+    template.id = generateRandomNumericId();
+    template.descr = generateRandomId();
+
+    cy.kcLogin('admin').as('tokens');
+  });
+
+  afterEach(() => cy.kcLogout());
+
+  it('Add content template', () => {
+    currentPage = openContentTemplatesPage();
+
+    cy.log(`Add content template with id ${template.id}`);
+    currentPage = currentPage.getContent().clickAddButton();
+    currentPage.getContent().typeId(template.id);
+    currentPage.getContent().typeName(template.descr);
+    currentPage.getContent().selectContentType(template.contentTypeText);
+    currentPage.getContent().typeHTMLModel(template.contentShape);
+    
+    currentPage = currentPage.getContent().submitForm();
+    currentPage.getContent().getTableRow(template.id).find(htmlElements.td).eq(0).should('contain.text', template.id);
+    currentPage.getContent().getTableRow(template.id).find(htmlElements.td).eq(2).should('contain.text', template.contentType);
+    currentPage.getContent().getTableRow(template.id).find(htmlElements.td).eq(4).should('contain.text', template.descr);
```

Comment:
You can collect all the validation in a single `then`, to avoid Cypress has to re-identify the table row each time